src/index.ts

```diff
@@ -1,43 +1,145 @@
-import {fetchTarball, getPackument, namedScope, prepareManifest, publish, scopedOptions} from './util'
-
-export async function sync(name: string, from: Record<string, string>, to: Record<string, string>, dryRun = false) {
-  // TODO: handle version spec
-  const scope = namedScope(name)
-  console.debug('scope', scope)
-
-  // get available source versions
-  const srcOptions = scopedOptions(scope, from.registry, from.token)
-  // fullMetadata may needed to obtain the repository property in manifest
-  srcOptions.fullMetadata = true
-  const srcPackument = await getPackument(name, srcOptions)
-  const srcVersions = srcPackument ? Object.keys(srcPackument.versions) : []
-  console.debug('Source versions', srcVersions)
-
-  // get available target versions
-  const dstOptions = scopedOptions(scope, to.registry, to.token)
-  const dstPackument = await getPackument(name, dstOptions)
-  const dstVersions = dstPackument ? Object.keys(dstPackument.versions) : []
-  console.debug('Target versions', dstVersions)
+import {ensureRepositoryAccess, fetchTarball, getScopedOptions, prepareManifest, publish} from './utils'
+import * as pacote from "pacote"
+import {filterSourceVersions} from "./filters"
+import {MigrationSettings} from "./settings"
```

Comment:
Codacy has a fix for the issue: [Replace `MigrationSettings}·from·"./settings"` with `·MigrationSettings·}·from·"./settings";`](https://app.codacy.com/gh/joebowbeer/regsync/pullRequest?prid=9263162)
```suggestion
import { MigrationSettings } from "./settings";
```